adapters/taboola/taboola.go

```diff
@@ -3,45 +3,66 @@ package taboola
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/prebid/openrtb/v17/native1"
+	nativeResponse "github.com/prebid/openrtb/v17/native1/response"
 	"github.com/prebid/openrtb/v17/openrtb2"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/config"
 	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
 	"github.com/prebid/prebid-server/openrtb_ext"
 	"net/http"
+	"net/url"
+	"text/template"
 )
 
 type adapter struct {
-	endpoint string
+	endpoint *template.Template
+	hostName string
 }
 
-// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+// Builder builds a new instance of Taboola adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+
+	hostName := ""
+	if server.ExternalUrl != "" {
+		parsedUrl, err := url.Parse(server.ExternalUrl)
+		if err == nil && parsedUrl != nil {
+			hostName = parsedUrl.Host
+		}
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
```

Comment:
`err` on line 26 gets shadowed by `err` on line 30. This function can be rephrased as follow

```
endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
if err != nil {
    return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
}

parsedUrl, err := url.Parse(server.ExternalUrl)
if err != nil {
     return nil, fmt.Errorf("unable to parse server external url: %v", err)
}

bidder := &adapter{
    endpoint: endpointTemplate,
    hostName: parsedUrl.Host,
}
return bidder, nil

```

Reply:
@onkarvhanumante  
added error return in case parse endpoint url failed. 

regarding the second check, it failed to extract hostName I pass empty hostName and don't want to fail the whole function. 

changes committed

Reply:
Just to confirm. With empty host final url looks like this: 
"https://display.bidder.taboola.com/OpenRTB/PS/auction//abc"
Is this ok to have `//`?

Reply:
@VeronikaSolovei9  yes - it's fine.